### PR TITLE
DNM  os/bluestore: adaptive throttle

### PR DESCRIPTION
**Pull request by liewegas**

### CODE REVIEW

File: src/os/bluestore/BlueStore.cc [hunk 1]
```diff
@@ -3314,11 +3328,43 @@ void BlueStore::handle_conf_change(const struct md_config_t *conf,
...
+    // reset internal throttle of the throttle manager
+    throttle_mgr.reset();
   }
   if (changed.count("bluestore_deferred_max_bytes")) {
     throttle_deferred_bytes.reset_max(
       conf->bluestore_max_bytes + conf->bluestore_deferred_max_bytes);
   }
+
+#if 0
+  // This cannot work as-is, because it will turn on throttle_ops,
```
Comment (amitkumar50) [inline src/os/bluestore/BlueStore.cc]: @liewegas Hello Sage. Its my comment that multiline comments should be inside /* */
Though these are internal not meant for customer facing docs but still 
We should maintain consistency over the code.

Comment (badone) [reply src/os/bluestore/BlueStore.cc]: @amitkumar50 OK, so suppose this is changed to /* */ What if I come along and want to comment out this whole #if/#endif, or half this file? How do I do this easily if all multi-line comments are in /**/ style? I'm not a fan of c-style comments.

**Comment by badone** [comment]
@liewegas What's the status of this PR? Is it still relevant?

**Comment by liewegas** [comment]
[bluestore] throttling is still key blocker for qos, and we don't have a solution yet, so I think this may still be relevant.

**Comment by badone** [comment]
@liewegas What's the status of this PR? Is it still fresh/relevant?

**Comment by badone** [comment]
@liewegas if you can resolve the conflicts I can review it...

**Comment by liewegas** [comment]
Well, what we really have to do is build a performance test environment so we can tell how well it works (and make it work better than any other options we come up with).

**Pull request closed without merging**